src/eko/io/metadata.py

```diff
@@ -65,11 +65,23 @@ def load(cls, path: os.PathLike):
         # read raw file first to catch version
         raw = yaml.safe_load(paths.metadata.read_text(encoding="utf-8"))
         version = parse(raw["version"])
+        data_version = raw["data_version"]
         # patch if necessary
-        if version.major == 0 and version.minor == 13:
-            raw = v1.update_metadata(paths, raw)
-        elif version.major == 0 and version.minor == 14:
-            raw = v2.update_metadata(paths, raw)
+        if data_version == 1:
+            if version.major == 0 and version.minor == 13:
+                raw = v1.update_metadata(paths, raw)
+            elif version.major == 0 and version.minor == 14:
+                raw = v2.update_metadata(paths, raw)
+            elif version.major == 0 and version.minor == 0 and "post" in raw["version"]:
```

Comment:
`raw` should not be accessed any longer at this point - use the object instead! (i.e. `version`) I'm sure you can do a similar trick there